runtime/src/iree-amd-aie/driver/xrt/direct_command_buffer.cc

```diff
@@ -0,0 +1,429 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/driver/xrt/direct_command_buffer.h"
+
+#include "iree-amd-aie/driver/xrt/native_executable.h"
+#include "iree-amd-aie/driver/xrt/pipeline_layout.h"
+#include "iree-amd-aie/driver/xrt/xrt_buffer.h"
+#include "iree/hal/utils/resource_set.h"
+
+typedef struct iree_hal_xrt_direct_command_buffer_t {
```

Comment:
This is a fair point, however I want to match C style in case I want to copy paste this backend and change the XRT API to some C API rather than a C++ API (which is the long term plan).